### PR TITLE
Stop on error

### PR DESCRIPTION
**Pull request by jkenny**

### CODE REVIEW

File: nova/core/deploy/__init__.py [hunk 1]
```diff
@@ -20,16 +20,20 @@
 
 STACK_ENV=`aws ec2 describe-tags --filters "Name=resource-id,Values=$INSTANCE_ID" "Name=key,Values=Environment" --region=$REGION --output=text | awk '{print $5}'`
 
-IFS=$'\r\n' GLOBIGNORE='*' command eval  'docker_args=($(cat /opt/nova/environments/$STACK_ENV/docker-vars.list))'
+IFS=$'\r\n' GLOBIGNORE='*' command eval  'docker_vars=($(cat /opt/nova/environments/$STACK_ENV/docker-vars.list))'
+
+eval DOCKER_OPTS=$(< /opt/nova/environments/$STACK_ENV/docker-opts.list)
```
Comment (fiadliel) [inline nova/core/deploy/__init__.py]: Do you need eval? In the past I would have considered

```
DOCKER_OPTS=$(cat /opt/nova/environments/$STACK_ENV/docker-opts.list)
```
but I guess `<` can replace cat in bash.

Comment (jkenny) [inline nova/core/deploy/__init__.py]: Without eval the text in the .list file is assigned to the variable without evaluating any commands contained in that text.

Comment (fiadliel) [inline nova/core/deploy/__init__.py]: 🤢 
But if that's the required behaviour, then okay, I guess.

File: nova/core/deploy/__init__.py [hunk 2]
```diff
@@ -20,16 +20,20 @@
...
+eval DOCKER_VOLS=$(< /opt/nova/environments/$STACK_ENV/docker-vols.list)
+eval DOCKER_ARGS=$(< /opt/nova/environments/$STACK_ENV/docker-args.list)
```
Comment (andreyk0) [inline nova/core/deploy/__init__.py]: I guess this works ... but seems to be a bit round about way of just sourcing a shell script with some variables defined.

```shell
#!/bin/sh

DOCKER_VOLUMES=...
DOCKER_OPTS=...
```

and to use it

```shell
#!/bin/sh

. /path/to/file/with/definitions

... the rest
```

Comment (andreyk0) [inline nova/core/deploy/__init__.py]: Oh, and to assert that a variable is defined, the usual pattern is to check if it's empty, e.g.:

```bash
#!/bin/bash

FOO=bar
#BLAH=blah

[ -z "$FOO" ] && { echo "Foo was not defined" >&2 ; exit 1; }
[ -z "$BLAH" ] && { echo "Blah was not defined" >&2 ; exit 1; }

echo running

```

Comment (jkenny) [inline nova/core/deploy/__init__.py]: @andreyk0 The .list files are generated by nova at deploy time based on the application's nova.yml configuration file. You're suggesting it would be better to generate a shell script instead and source it in the main script?

Comment (andreyk0) [inline nova/core/deploy/__init__.py]: Of course! :)
You are already generating *something*, why not generate exactly what you want?

Comment (jkenny) [inline nova/core/deploy/__init__.py]: @andreyk0 You mean something like this? https://github.com/gilt/nova/commit/9ff2dacb93c1e29984aeba082b33dcc9ebf3ae58

Comment (andreyk0) [inline nova/core/deploy/__init__.py]: Sort of .. I'd get rid of all evals and .list files, would have python code generate a shell script with all variables defined that can then be sourced into your startup script.

Comment (jkenny) [inline nova/core/deploy/__init__.py]: @andreyk0 I'm moving this suggestion to a separate PR: https://github.com/gilt/nova/pull/24

Comment (andreyk0) [inline nova/core/deploy/__init__.py]: Cool, commented there.
